skimage/_shared/utils.py

```diff
@@ -726,6 +726,8 @@ def _supported_float_type(input_dtype, allow_complex=False):
     float_type : dtype
         Floating-point dtype for the image.
     """
+    if hasattr(input_dtype, 'dtype'):
+        input_dtype = input_dtype.dtype
     if isinstance(input_dtype, Iterable) and not isinstance(input_dtype, str):
         return np.result_type(*(_supported_float_type(d) for d in input_dtype))
     input_dtype = np.dtype(input_dtype)
```

Comment:
The test's fail with `TypeError: Cannot interpret '<attribute 'dtype' of 'numpy.generic' objects>' as a data type`. I think this should do
```suggestion
    if hasattr(input_dtype, 'dtype'):
        input_dtype = input_dtype.dtype
    elif isinstance(input_dtype, Iterable) and not isinstance(input_dtype, str):
        return np.result_type(*(_supported_float_type(d) for d in input_dtype))
    else:
        input_dtype = np.dtype(input_dtype)
```

Reply:
This caused more issues (AttributeError: 'getset_descriptor' object has no attribute 'char')
I used a try/except instead. 

I try casting `input_dtype` with `input_dtype = np.dtype(input_dtype)` and I then catch two exceptions:

- `TypeError` for cases when an array is passed as argument (`TypeError: Cannot construct a dtype from an array`). 
- ValueError when an array with two different float values

Reply:
This caused more issues (AttributeError: 'getset_descriptor' object has no attribute 'char')
I used a try/except instead. 

I try casting `input_dtype` with `input_dtype = np.dtype(input_dtype)`. This works in cases where input_dtype is a numpy dtype. I then catch two exceptions:

- `TypeError` for cases when an array is passed as argument (`TypeError: Cannot construct a dtype from an array`). This is then handled in the `Except` block with `getattr(input_dtype, 'dtype', input_dtype)`

- ValueError for cases where `input_dtype` is an array with two different float values. I had this issue in the `test_supported_float_dtype_sequence` test where _supported_float_type((np.float16, np.float32)) is tested

The tests look fine now